actions/retest/main.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2021 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/google/go-github/github"
+	"golang.org/x/oauth2"
+)
+
+// standard input env variables in action more details at
+// https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputs
+var (
+	retryLimit          = os.Getenv("INPUT_MAX-RETRY")
```

Comment:
It would be cleaner to have a type `retestConfig`, fill that like you do now, and call a `retestConfig.validate()` function so that you can split validation of all the variables in a separate function.